pkg/scheduler/scheduler.go

```diff
@@ -297,7 +297,7 @@ func (sched *Scheduler) preempt(preemptor *v1.Pod, scheduleErr error) (string, e
 	if !util.PodPriorityEnabled() || sched.config.DisablePreemption {
 		klog.V(3).Infof("Pod priority feature is not enabled or preemption is disabled by scheduler configuration." +
 			" No preemption is performed.")
-		return "", nil
+		return "", fmt.Errorf("Pod priority feature is not enabled or preemption is disabled.")
```

Comment:
I know it isn't error. The error returned here is just to tell scheduler.scheduleOne not to deal with the metrics data, because we do nothing about the preempt when util.PodPriority disable or sched.config.DisablePreemption. Actually, I think the judgement is best placed outside of sched.preempt.

Reply:
Another way to avoid metrics being handled, you can add the `if !util.PodPriorityEnabled() || sched.config.DisablePreemption` before calculating the metrics. @ping035627

Reply:
So add judgement of error returned by `shced.preempt` with `if !util.PodPriorityEnabled() || sched.config.DisablePreemption`?

Reply:
yes

Reply:
As other said, this shouldn't be an error. Errors appear in our logs and can be confusing. Preemption being disabled is not an error.